lib/Doctrine/ORM/Query.php

```diff
@@ -250,7 +265,7 @@ private function _parse()
             return $this->_parserResult;
         }
 
-        $hash   = $this->_getQueryCacheId();
+        $hash   = $this->_getQueryCacheId($types);
```

Comment:
For consistency with the rest of the class, I'd say that `_getQueryCacheId` directly accesses `$this->_parsedTypes()` instead, no?